### ISSUE TITLE
Rule no-stateless-class does not account for constructors using parameter properties

### ISSUE THREAD

**Issue by rsaddey** [Microsoft/tslint-microsoft-contrib#310]
```
class Point {  
    constructor(public x: number, public y: number) {  
        // Constructor body  
    }  
}
```
has 2 private public properties but is still marked as stateless.

See [TypeScript - Handbook - Classes: Parameter Properties](http://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties)
and [TypeScript Language Specification (v1.8.2) at 8.3.1 Constructor Parameters](https://github.com/Microsoft/TypeScript/blob/v1.8.2/doc/spec.md#831-constructor-parameters)

**Comment by HamletDRC** [comment]
The modifiers available in 2.0 are: 
* public
* private
* protected
* readonly

**Comment by HamletDRC** [comment]
Thanks for the bug report! 

The fix is available on our "releases" branch. So you can point your npm to that github branch and start using the fix today if you'd like.
